examples/src/main/scala/za/co/absa/fadb/examples/enceladus/DatasetSchema.scala

```diff
@@ -60,72 +59,63 @@ object DatasetSchema {
+    override protected def sql(values: SchemaInput): SQLActionBuilder = {
       sql"""SELECT A.status, A.status_text, A.id_schema_version
-             FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
-                ${values.fields}::JSONB, ${values.userName}
-             ) A;"""
+            FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
+              ${values.fields}::JSONB, ${values.userName}
+            ) A;"""
     }
 
-    override protected def resultConverter: GetResult[Long] = {
-      val gr:GetResult[Long] = GetResult(r => {
-        val status: Int = r.<<
-        val statusText: String = r.<<
-        if (status != 201) throw new StatusException(status, statusText)
-        r.<<
-      })
-      gr
-    }
+    override protected def slickConverter: GetResult[Long] = GetResult.GetLong
+
+    override def OKStatuses: Set[Integer] = Set(201)
+
   }
 
-  final class GetSchema(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBUniqueFunction[ExecutorEngineType, (String, Option[Int]), Schema](schema)
-    with SlickPgFunction[(String, Option[Int]), Schema] {
+  final class GetSchema(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBUniqueFunction[(String, Option[Int]), Schema, SlickPgEngine]
+    with SlickPgFunctionWithStatusSupport[(String, Option[Int]), Schema]
+    with UserDefinedStatusHandling {
 
     def apply(id: Long): Future[Schema] = {
       val sql =
         sql"""SELECT A.*
              FROM #$functionName($id) A;"""
 
-      schema.unique(makeQueryFunction(sql)(resultConverter))
+      val slickQuery = new dBEngine.QueryType(sql, slickConverter)
+      dBEngine.unique[Schema](slickQuery)
     }
 
-    override protected def sqlToCallFunction(values: (String, Option[Int])): SQLActionBuilder = {
+    override protected def sql(values: (String, Option[Int])): SQLActionBuilder = {
       sql"""SELECT A.*
-              FROM #$functionName(${values._1}, ${values._2}) A;"""
+            FROM #$functionName(${values._1}, ${values._2}) A;"""
+    }
+
+    override protected val slickConverter: GetResult[Schema] = GetResult{r =>
+      Schema(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
     }
 
-    override protected def resultConverter: GetResult[Schema] = DatasetSchema.GetSchemaImplicit
+    override val OKStatuses: Set[Integer] = Set(200)
   }
 
-  final class ListSchemas(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBSeqFunction[ExecutorEngineType, Boolean, SchemaHeader](schema)
+  final class List(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBSeqFunction[Boolean, SchemaHeader, SlickPgEngine]()
     with SlickPgFunction[Boolean, SchemaHeader] {
 
     override def apply(values: Boolean = false): Future[Seq[SchemaHeader]] = super.apply(values)
```

Comment:
Perhaps you just wanted to demonstrate the possibilities with this `apply` here? Because otherwise I don't think it's necessary - I can just go through the class hierarchy and substitute `R` and `I` into parents and I got the same information basically as it is here on this line

Reply:
Yes, it's because I am adding a default value to the input parameter. 👍

Reply:
Ok, np, thanks :)